camp/camp-brooklyn/src/main/java/org/apache/brooklyn/camp/brooklyn/spi/dsl/methods/DslComponent.java

```diff
@@ -340,9 +360,9 @@ public DslConfigSupplier(DslComponent component, String keyName) {
         public final Maybe<Object> getImmediately() {
             Maybe<Entity> targetEntityMaybe = component.getImmediately();
             if (targetEntityMaybe.isAbsent()) return Maybe.absent("Target entity not available");
-            Entity targetEntity = targetEntityMaybe.get();
+            EntityInternal targetEntity = (EntityInternal) targetEntityMaybe.get();
             
-            return Maybe.of(targetEntity.getConfig(ConfigKeys.newConfigKey(Object.class, keyName)));
+            return targetEntity.config().getNonBlocking(ConfigKeys.newConfigKey(Object.class, keyName));
```

Comment:
Yes, I'd updated `AbstractConfigurationSupportInternal` line 142 to use `.immediately(true)`, so I hope that covers the case of `getNonBlockingResolvingSimple`. I didn't want to touch the "structure key" code. We should probably change that at some point to go through ValueResolver, even if that still ends up scheduling it in a task.

I therefore think this comment has been addressed. Am I missing something?